### PR TITLE
Added support for matrix parameters.

### PR DESCRIPTION
**Pull request by dlarsson**
HasJQ instances for matrix parameter types. When adding the segmentToStr function, I got "unresolved" errors calling jsMParams. If I moved segmentToStr below the makeLenses calls, it worked. So I moved all functions down below.

The Segment type is slightly refactored, as both captures and static segments may have matrix parameters.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,17 @@
+dist
+cabal-dev
+*.o
```
Comment (christian-marie) [inline .gitignore]: cabal-sandbox things and dist seem reasonable, but there's some very odd things in this .gitignore that should probably be in your local config.

**Comment by christian-marie** [comment]
Looks good, can we get some tests for the added functionality and a merge against latest upstream?

**Comment by jkarni** [comment]
@dlarsson any chance you could address the issues mentioned (`servant-jquery` is the only package missing a release with support for matrix params).

**Comment by dlarsson** [comment]
Ooops, sorry I missed this. Sure, I can do that. I'm at work, and the code
is on my home computer, so can't fix it right away. I think the original
.gitignore was something I found somewhere on github as a "default when
using haskell", and I admit I didn't analyze it too thoroughly. I'll check
it this evening

2015-03-02 15:42 GMT+01:00 Julian Arni <notifications@github.com>:

**Comment by christian-marie** [comment]
It's mostly the rebase and a basic regression test. I think we can live with the gitignore :)

**Comment by alpmestan** [comment]
We need this merged for 0.3, since matrix params have made it into releases for other packages.

**Pull request closed without merging**